lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMMAv2.cpp

```diff
@@ -584,55 +608,175 @@ Value loadArg(ConversionPatternRewriter &rewriter, Location loc, Value tensor,
+  if (auto sharedEncoding = encoding.dyn_cast<SharedEncodingAttr>()) {
+    auto order = sharedEncoding.getOrder();
+    auto rank = order.size();
+    if (rank == 3) {
+      return encoding;
+    }
+    auto expandedOrder = SmallVector<unsigned>(3, 0);
+    expandedOrder[0] = order[0] + 1;
+    expandedOrder[1] = order[1] + 1;
+    ArrayRef<unsigned> expandedOrderArr(expandedOrder);
+    auto expandedEncoding = SharedEncodingAttr::get(
+        ctx, sharedEncoding.getVec(), sharedEncoding.getPerPhase(),
+        sharedEncoding.getMaxPhase(), expandedOrderArr,
+        getExpandedCTALayout(ctx, sharedEncoding.getCTALayout()),
+        sharedEncoding.getHasLeadingOffset());
+    return expandedEncoding;
+  } else if (auto mmaEncoding = encoding.dyn_cast<NvidiaMmaEncodingAttr>()) {
+    auto warpsPerCTA = triton::gpu::getWarpsPerCTA(mmaEncoding);
+    auto rank = warpsPerCTA.size();
+    if (rank == 3) {
+      return encoding;
+    }
+    auto expandedWarpsPerCTA = insertValue<unsigned>(warpsPerCTA, 0, 1);
+    auto instrShape = mmaEncoding.getInstrShape();
+    auto expandedInstrShape = insertValue<unsigned>(instrShape, 0, 1);
+    auto expandedMmaEncoding = NvidiaMmaEncodingAttr::get(
+        ctx, mmaEncoding.getVersionMajor(), mmaEncoding.getVersionMinor(),
+        expandedWarpsPerCTA,
+        getExpandedCTALayout(ctx, mmaEncoding.getCTALayout()),
+        expandedInstrShape);
+    return expandedMmaEncoding;
+  } else if (auto dotOperandEncoding =
+                 encoding.dyn_cast<DotOperandEncodingAttr>()) {
+    auto mmaEncoding =
+        dotOperandEncoding.getParent().cast<NvidiaMmaEncodingAttr>();
+    auto expandedMMAEncoding = getExpandedEncoding(mmaEncoding);
+    auto expandedEncoding = DotOperandEncodingAttr::get(
+        ctx, dotOperandEncoding.getOpIdx(), expandedMMAEncoding,
+        dotOperandEncoding.getKWidth());
+    return expandedEncoding;
+  } else
+    llvm_unreachable("unsupported encoding");
+}
+
+RankedTensorType getExpandedTensor(RankedTensorType tensorTy) {
+  auto shapePerCTA = getShapePerCTA(tensorTy);
+  auto rank = shapePerCTA.size();
+  if (rank == 3)
+    return tensorTy;
+
+  auto elTy = tensorTy.getElementType();
+  auto shape = tensorTy.getShape();
+  auto expandedShape = SmallVector<int64_t>(3, 1);
+  expandedShape[1] = shape[0];
+  expandedShape[2] = shape[1];
+  auto encoding = tensorTy.getEncoding();
+  auto expandedEncoding = getExpandedEncoding(encoding);
+  auto expandedTensor =
+      RankedTensorType::get(expandedShape, elTy, expandedEncoding);
+  return expandedTensor;
+}
+
+SharedMemoryObject
+getExpandedSharedMemoryObject(ConversionPatternRewriter &rewriter, Location loc,
+                              SharedMemoryObject smemObj,
+                              ArrayRef<int64_t> shape) {
+  auto strides = smemObj.getStrides();
+  auto offsets = smemObj.getOffsets();
+  auto rank = strides.size();
+  if (rank == 3)
+    return smemObj;
+  // TODO: is this wrong ?
```

Comment:
Why it might be wrong?

Reply:
The comment is wrong, forgot to remove it